### PR TITLE
Fixup @mentions 

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Click catcher!
- [x] Better styling (layout, Usernames component)!
- [x] Tabs to move
- [x] Wrap around selection
- [x] Keep selection in view
- [x] Update text when selection changes

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers 
cc: @MarcoPolo

File: shared/chat/conversation/input/index.desktop.js [hunk 1]
```diff
@@ -133,7 +142,7 @@ class ConversationInput extends Component<InputProps, State> {
       this.setState(({upArrowCounter}) => ({upArrowCounter: upArrowCounter + 1}))
     }
 
-    if (e.key === 'ArrowDown' && this.props.mentionPopupOpen) {
+    if (['ArrowDown', 'Tab'].includes(e.key) && this.props.mentionPopupOpen) {
```
Comment (buoyad) [inline shared/chat/conversation/input/index.desktop.js]: I've seen tab more often used for completion (bash, slack, facebook, etc.)

**Comment by buoyad** [comment]
LGTM, tested and it's really usable 👍

**Comment by chrisnojima** [comment]
I based this in slack which changes like this does

**Comment by buoyad** [comment]
Ah, I never noticed that: slack does both. Moves down when you only have `@` input and autocompletes if you've already written some username. ¯\_(ツ)_/¯

**Comment by chrisnojima** [comment]
Ah. Man this component is more complicated than I thought ;) I think making
it work like that is 1 more like of code. Lemme see if it’s trivial to add

**Comment by chrisnojima** [comment]
ok @buoyad PTAL. This adds that tab handling , plus shift+tab, plus escape

**Comment by buoyad** [comment]
Feels better w/r/t the tabs & esc working, it can get the input into a weird state when you cursor left/right through an @-mention while the screen is up. I think it can be fixed by making `_getWordAtCursor` stop at the first non-username character (or the end of the input) rather than literally where the cursor is

**Comment by chrisnojima** [comment]
turns out there is a flag that does this but, looking at slack etc they just close the dialog on left/right since it raises a bunch of ux issues. so lets just do that

**Comment by chrisnojima** [comment]
@buoyad ptal

**Comment by buoyad** [comment]
LGTM 👍

**Pull request merged**